README.md

```diff
@@ -812,150 +1113,292 @@ type Firework = record {
 };
 ```
 
-#### Array
+#### func
 
-TypeScript `[]` array syntax corresponds to the [Candid type vec](https://smartcontracts.org/docs/candid-guide/candid-types.html#type-vec) and will become an array of the enclosed type at runtime. Only the `[]` array syntax is supported at this time (i.e. not `Array` or `ReadonlyArray` etc).
+The Azle type `func` corresponds to the [Candid type func](https://internetcomputer.org/docs/current/references/candid-ref/#type-func---) and at runtime will become a JavaScript array with two elements, the first being an [@dfinity/principal](https://www.npmjs.com/package/@dfinity/principal) and the second being a [JavaScript string](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String). The `@dfinity/principal` represents the `principal` of the canister/service where the function exists, and the `string` represents the function's name.
 
 TypeScript:
 
 ```typescript
-import { Query, int32 } from 'azle';
+import { Func, nat64, Principal, Query, Update, Variant } from 'azle';
 
-export function getNumbers(): Query<int32[]> {
-    return [0, 1, 2, 3];
+type User = {
+    id: string;
+    basic_func: BasicFunc;
+    complex_func: ComplexFunc;
+};
+
+type Reaction = Variant<{
+    Good: null;
+    Bad: null;
+    BasicFunc: BasicFunc;
+    ComplexFunc: ComplexFunc;
+}>;
+
+type BasicFunc = Func<(param1: string) => Query<string>>;
+type ComplexFunc = Func<(user: User, reaction: Reaction) => Update<nat64>>;
+
+export function get_basic_func(): Query<BasicFunc> {
+    return [
+        Principal.fromText('rrkah-fqaaa-aaaaa-aaaaq-cai'),
+        'simple_function_name'
+    ];
+}
+
+export function get_complex_func(): Query<ComplexFunc> {
+    return [
+        Principal.fromText('ryjl3-tyaaa-aaaaa-aaaba-cai'),
+        'complex_function_name'
+    ];
 }
 ```
 
 Candid:
 
 ```typescript
-service: {
-    "getNumbers": () -> (vec int32) query;
+type User = record {
+    "id": text;
+    "basic_func": BasicFunc;
+    "complex_func": ComplexFunc;
+};
+type Reaction = variant { "Good": null; "Bad": null; "BasicFunc": BasicFunc; "ComplexFunc": ComplexFunc };
+
+type BasicFunc = func (text) -> (text) query;
+type ComplexFunc = func (User, Reaction) -> (nat64);
+
+service: () -> {
+    "get_basic_func": () -> (BasicFunc) query;
+    "get_complex_func": () -> (ComplexFunc) query;
 }
+
 ```
 
-#### Opt
+#### service
 
-The Azle type `Opt` corresponds to the [Candid type opt](https://smartcontracts.org/docs/candid-guide/candid-types.html#type-opt) and will become the enclosed JavaScript type or null at runtime.
+Not yet implemented.
```

Comment:
I'd recommend linking to the issue (445) here so that people can get more context around when it's coming (like, see what milestone it's in and such).